metrics/config.go

```diff
@@ -74,23 +77,17 @@ type metricsConfig struct {
 	// reportingPeriod specifies the interval between reporting aggregated views.
 	// If duration is less than or equal to zero, it enables the default behavior.
 	reportingPeriod time.Duration
+	
+	// recorder provides a hook for performing custom transformations before
+	// writing the metrics to the stats.RecordWithOptions interface.
+	recorder func(context.Context, stats.Measurement, ...stats.Options) error
```

Comment:
Should the recorder func be parented to the metrics pkg itself and be passed the metricsConfig as an input rather than being a part of the metricsConfig? Similar to how kubernetes/client-go/metrics pkg does it (go/blob/03bfb9bdcfe5482795b999f39ca3ed9ad42ce5bb/tools/metrics/metrics.go)?

Limiting metricsConfig to just user-configurable state seems like the ideal state. Stubbing out core functionality of the metrics pkg feels like an overscope of a config struct.

Reply:
I agree with this criticism; this is not the end state, but step 1 of N.

If you want, I can try to do this in bigger chunks. I knew you were working in this general area, and I didn't want to make too many disruptive changes at once (as it was, I got a merge on the rename of `stackdriverClientConfig` -> `StackdriverClientConfig`).

Reply:
Splitting out the changes into smaller sets makes sense to me. Just wanted to clarify where the end state of metricsConfig was going.

Especially when you know I'm about to cause merge conflicts :)